web/src/pages/CreateRoom/CreateRoom.tsx

```diff
@@ -0,0 +1,109 @@
+import { Component, createSignal, Show } from "solid-js";
+import Button from "@/components/Button";
+
+import styles from "./CreateRoom.module.scss";
+import useCreateRoom from "./hooks/useCreateRoom";
+
+const prohibitedEmojiHex = "0x1F6AB";
+
+const CreateRoom: Component = () => {
+	const { fields, updateField, submit } = useCreateRoom();
+
+	const handleSubmit = (event: Event): void => {
+		event.preventDefault();
+		submit();
+	};
+
+	return (
+		<main class={styles.createRoom}>
+			<form class={styles.createRoomForm} onSubmit={handleSubmit}>
+				<label for="moderatorName">
+					Your name
+					<span class={styles.required}>*</span>
+				</label>
+
+				<input
+					name="moderatorName"
+					required
+					type="text"
+					onInput={updateField}
+				/>
+
+				<div class={styles.seperator}>
+					<hr />
+				</div>
+
+				<label for="voterOptions">
+					What options would you like voters have to choose from?
+					<span class={styles.required}>*</span>
+				</label>
+
+				<select name="voterOptions" required onInput={updateField}>
+					<option value="">Select options</option>
+					<option value="fibonacci">Fibonacci</option>
+					<option value="linear">Linear</option>
+				</select>
+
+				<label for="numberOfOptions">Number of options</label>
+
+				<input
+					type="range"
+					name="numberOfOptions"
+					min="2"
+					max="15"
+					value={fields.numberOfOptions}
+					step="1"
+					disabled={fields.disabled}
+					onInput={updateField}
+				/>
+
+				<label for="noVote">Include no-vote option?</label>
+
+				<label class={styles.radio} for="noVote">
```

Comment:
Clicking the labels doesn't select the option. Are your labels correctly tied to the inputs? Also, should these inputs be in a fieldset? That way you can handle the onInput on the fieldset instead of individually on the radios.